services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java

```diff
@@ -990,6 +990,14 @@ private void updateFederatedIdentity(BrokeredIdentityContext context, UserModel
 
         if (context.getIdpConfig().getSyncMode() == IdentityProviderSyncMode.FORCE) {
             setBasicUserAttributes(context, federatedUser);
+
+            if (!Objects.equals(context.getUsername(), federatedIdentityModel.getUserName())) {
```

Comment:
@danielFesenmeyer Do you think it makes sense to have a check for the subject and make sure the link being created still references the same subject?

I don't expect this check to fail though because the subject should not change (not 100% sure though if the brokered OP is using pairwise).

Reply:
@pedroigor: As I understand, a check for the subject is not necessary here, because it is used to find the federated identity after IDP authentication (context.getId() is equal to subject in case of OIDC): 

https://github.com/keycloak/keycloak/blob/8d5c11a4a551a5267bac4942da8d21e2867bd98c/services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java#L526
...
https://github.com/keycloak/keycloak/blob/8d5c11a4a551a5267bac4942da8d21e2867bd98c/services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java#L534
...
https://github.com/keycloak/keycloak/blob/f69497eb28ab6c6595a4a335607d03b0c7ba229d/model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java#L552

So the method IdentityBrokerService#updateFederatedIdentity(...) will not be called in case the subject changed.